src/SmartComponents/DriftPage/DriftTable/ComparisonHeader/ComparisonHeader.js

```diff
@@ -12,33 +12,44 @@ import { ASC, DESC } from '../../../../constants';
 import HistoricalProfilesPopover from '../../../HistoricalProfilesPopover/HistoricalProfilesPopover';
 import ReferenceSelector from '../ReferenceSelector/ReferenceSelector';
 
-class ComparisonHeader extends Component {
-    constructor(props) {
-        super(props);
-        this.columnWidth = React.createRef();
-
-        this.state = {
-            refState: null
-        };
-    }
-
-    setColumnWidth = () => {
-        if (this.columnWidth?.current) {
-            this.props.setColumnHeaderWidth(this.columnWidth.current.offsetWidth);
-            this.setState({ refState: this.columnWidth });
+const ComparisonHeader = ({
+    factSort,
+    fetchCompare,
+    masterList,
+    permissions,
+    referenceId,
+    removeSystem,
+    selectedBaselineIds,
+    selectedHSPIds,
+    selectHistoricProfiles,
+    setColumnHeaderWidth,
+    setHistory,
```

Comment:
This `setHistory` is passed very deep here, as are other callback functions...

While converting Drift to function components it might be good to take a look at these and see if what they do can be done without passing them done.

For example, in the case of `setHistory`, which triggers a `navigate` call, we could introduce a `useSetHistory` hook to import and use here instead.

Reply:
I just realised this is exactly what @adonispuente is doing here: https://github.com/RedHatInsights/drift-frontend/pull/673

Reply:
If we want to merge #673 first, then I can implement it. @adonispuente in case there's more you need to do with it. Seems like it needs to be removed from draft and Sebastian can review again.

Reply:
Is this conversation resolved ? 
@johnsonm325 @bastilian

Reply:
Yeah, this is mure for now. But we'll want to look at the passing of setHistory at some point.